components/post_view/post_body/post_body.tsx

```diff
@@ -17,89 +20,93 @@ import PostBodyAdditionalContent from 'components/post_view/post_body_additional
 import PostMessageView from 'components/post_view/post_message_view';
 import ReactionList from 'components/post_view/reaction_list';
 import LoadingSpinner from 'components/widgets/loading/loading_spinner';
+import {PluginsState} from 'types/store/plugins';
 
 const SENDING_ANIMATION_DELAY = 3000;
 
-export default class PostBody extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * The post to render the body of
-         */
-        post: PropTypes.object.isRequired,
-
-        /**
-         * The parent post of the thread this post is in
-         */
-        parentPost: PropTypes.object,
-
-        /**
-         * The poster of the parent post, if exists
-         */
-        parentPostUser: PropTypes.object,
-
-        /**
-         * The function called when the comment icon is clicked
-         */
-        handleCommentClick: PropTypes.func.isRequired,
-
-        /**
-         * Set to render post body compactly
-         */
-        compactDisplay: PropTypes.bool,
-
-        /**
-         * Set to highlight comment as a mention
-         */
-        isCommentMention: PropTypes.bool,
-
-        /**
-         * Set to render a preview of the parent post above this reply
-         */
-        isFirstReply: PropTypes.bool,
-
-        /*
-         * Post type components from plugins
-         */
-        pluginPostTypes: PropTypes.object,
-
-        /**
-         * Flag passed down to PostBodyAdditionalContent for determining if post embed is visible
-         */
-        isEmbedVisible: PropTypes.bool,
-
-        /**
-         * Whether or not the post username can be overridden.
-         */
-        enablePostUsernameOverride: PropTypes.bool.isRequired,
-
-        /**
-         * Set not to allow edits on post
-         */
-        isReadOnly: PropTypes.bool,
-    }
-
+type Props = {
+
+    /**
+   * The post to render the body of
+   */
+    post: Post;
+
+    parentPost?: Post;
+
+    /**
+   * The parent post of the thread this post is in
+   */
+    parentPostUser?: UserProfile | null;
```

Comment:
Are we checking for `null` specifically at some point for this property? I couldn't find it myself.

Reply:
If I will not put this then compiler start complaining in the `index.ts` file. 

![image](https://user-images.githubusercontent.com/351564/100123433-8111bb00-2ea0-11eb-8bfc-139ab2fce601.png)

Reply:
This is the error -->

`Types of parameters 'props' and 'nextProps' are incompatible.
          Type 'Readonly<Matching<{ parentPost: Post | undefined; parentPostUser: UserProfile | null | undefined; pluginPostTypes: { [postType: string]: PostPluginComponent; }; enablePostUsernameOverride: boolean; isReadOnly: boolean; } & DispatchProp<...>, ClassAttributes<...> & Props>>' is not assignable to type 'Props'.
            Types of property 'parentPostUser' are incompatible.
              Type 'UserProfile | null | undefined' is not assignable to type 'UserProfile | undefined'.
                Type 'null' is not assignable to type 'UserProfile | undefined'.ts(2345)`

Reply:
Ahh, it's because of this line in index.js:
`parentPostUser = parentPost ? getUser(state, parentPost.user_id) : null;`

We need to change that `null` to `undefined` to fix that.